cmake/DetectCodeCoverage.cmake

```diff
@@ -20,7 +26,8 @@ if ( GCOV AND
      LCOV AND
```

Comment:
I am not sure if it is a good idea to always in inject `-O0` if these tools are found, as it makes the code slow.

I use Travis with [codecov](codecov.io), which just a parses all the gcov output files.